proto/google/events/firebase/v1/data.proto

```diff
@@ -17,22 +17,327 @@ syntax = "proto3";
 package google.events.firebase.v1;
```

Comment:
General comment: I don't think I like having all of the Firebase events in a single file. I would recommend a structure similar to Cloud, with each provider having its own separate directory and lifecycle. While I'm helping to shepherd these holistically at the moment, in the future I'd expect each Firebase team to maintain them independently.

Reply:
Splitting Firebase events into separate protos would make sense.

I'm not sure what's the CE `source` and the main serviceName for these events, but that could factor in the split up name.

For example: `identitytoolkit.googleapis.com`, `firestore.googleapis.com`

Reply:
I've got a doc with proposals for this (which I need to share more widely soonish) but my recollection is that the source varies more widely than is useful for the protobuf package naming here. I think keeping everything under `firebase` will actually be more intuitive.
I think what I might do is leave this existing PR for comparison, but create a new PR with the split version.